packages/block-editor/src/components/link-control/index.js

```diff
@@ -367,7 +426,9 @@ function LinkControl( {
 							variant="primary"
 							onClick={ handleSubmit }
 							className="block-editor-link-control__search-submit"
-							disabled={ currentInputIsEmpty } // Disallow submitting empty values.
+							disabled={
+								! valueHasChanges || currentInputIsEmpty
+							}
```

Comment:
This means the `Apply` button will not be active until changes to the internal value are detected.